### ISSUE TITLE
saxs, splitpixel mode: errors are over-estimated (patch included)

### ISSUE THREAD

**Issue by dfranke76** [silx-kit/pyFAI#883]
pyFAI in splitpixel mode severely over-estimates the errors. The figure below was compiled from pair-wise comparisons of 1390 consecutive SAXS frames of water obtained at P12 at PETRA-III. The frames are similar up to noise as per cormap test as implemented in datcmp of ATSAS, therefore any deviations from the expected chi^2 distribution should be from the error estimates.

![chi2-pval](https://user-images.githubusercontent.com/15729164/43761969-f33a8c5e-9a26-11e8-9d91-1e419d03fdbe.png)

Panel (a) shows the empirical distribution of chi^2 values of pyFAI in splitpixel mode, panel (b) the corresponding p-values. Note that the distribution in (a) is far away from the expected location at 1.0, the p-values are all 1.0 as the expected distribution is to the right of the empirical distribution (a uniform distribution of p-values in [0,1] would be expected).

Panel (c) shows the empirical distribution of chi^2 values after application of the suggested changes quoted below. Now the empirical distribution appears to be much closer to the expected one, but now seems to exhibit the same under-estimation of errors as the histogram mode (issue  #882).

Suggested patch (use the standard error of the weighted mean, not the standard error of the mean):

```
Index: pyFAI/ext/splitPixel.pyx
===================================================================
--- pyFAI/ext/splitPixel.pyx    (revision 6562)
+++ pyFAI/ext/splitPixel.pyx    (working copy)
@@ -155,6 +155,7 @@
         position_t[:, :, ::1] cpos = numpy.ascontiguousarray(pos, dtype=position_d)
         data_t[::1] cdata = numpy.ascontiguousarray(weights.ravel(), dtype=data_d)
         acc_t[::1] sum_data = numpy.zeros(bins, dtype=acc_d)
+        acc_t[::1] sum_data_squared = numpy.zeros(bins, dtype=acc_d)
         acc_t[::1] sum_count = numpy.zeros(bins, dtype=acc_d)
         data_t[::1] merged = numpy.zeros(bins, dtype=data_d)
         mask_t[::1] cmask
@@ -324,6 +325,7 @@
                     sub_area = sub_area * one_over_area
                     sum_count[i] += sub_area
                     sum_data[i] += sub_area * data
+                    sum_data_squared[i] += (sub_area ** 2) * data
 
                 # Check the total area:
                 if fabs(sum_area - area_pixel) / area_pixel > 1e-6 and bin0_min != 0 and bin0_max != bins:
@@ -340,7 +342,7 @@
                                  pos0_max - 0.5 * dpos, 
                                  bins)
 
-    return (bin_centers, numpy.asarray(merged), numpy.asarray(sum_data), numpy.asarray(sum_count))
+    return (bin_centers, numpy.asarray(merged), numpy.asarray(sum_data_squared), numpy.asarray(sum_count))
 
 
 @cython.cdivision(True)
```

**Comment by kif** [comment]
Hi Daniel,

I agree with you patch. As this code is used for processing the mean and the standard deviation, I will probably better add an options for selecting the power for the coeficient to be used.

**Comment by kif** [comment]
Fixed in version 0.20.
